packages/grafana-ui/src/components/Alert/Alert.tsx

```diff
@@ -1,21 +1,49 @@
 import React, { FC, ReactNode } from 'react';
+import classNames from 'classnames';
 
-interface Props {
+export enum AlertVariant {
+  Success = 'success',
+  Warning = 'warning',
+  Error = 'error',
+  Info = 'info',
+}
+
+interface AlertProps {
   title: string;
   button?: {
-    text: string;
+    text?: string;
```

Comment:
I would change this nested object form to buttonText?, onButtonClick?, onRemove?  if onRemove is specified it should be rendered like this
![image](https://user-images.githubusercontent.com/10999/65220632-60d7b680-dabb-11e9-9c49-8045722941d5.png)

Reply:
Thanks torkleo. If both buttonText and onRemove is specified, shall we give preference to onRemove and show the remove icon only or show icon along with buttonText?

Reply:
yes!

Reply:
I have updated the PR. Please take a look